File: scripts/run_pipeline_test.js
const http = require('http');
function post(path, payload) {
  return new Promise((resolve, reject) => {
    const data = JSON.stringify(payload);
    const options = {
      hostname: '127.0.0.1',
      port: 8000,
      path,
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Content-Length': Buffer.byteLength(data)
      }
    };
    const req = http.request(options, res => {
      let body = '';
      res.on('data', chunk => body += chunk);
      res.on('end', () => resolve({ status: res.statusCode, body }));
    });
    req.on('error', reject);
    req.write(data);
    req.end();
  });
}

(async () => {
  try {
    console.log('Triggering training...');
    const t = await post('/v1/generate/train', { n_candidates: 6 });
    console.log('train', t.status, t.body);

    console.log('\nCalling predict...');
    const p = await post('/v1/predict/', { smiles_list: ['CCO', 'c1ccccc1'] });
    console.log('predict', p.status, p.body);
  } catch (e) {
    console.error('error', e);
    process.exit(1);
  }
})();

File: scripts/run_pipeline_test.py
import urllib.request, json, sys

def post(url, payload):
    data = json.dumps(payload).encode()
    req = urllib.request.Request(url, data, headers={'Content-Type': 'application/json'})
    with urllib.request.urlopen(req) as r:
        return r.status, r.read().decode()

if __name__ == '__main__':
    base = 'http://127.0.0.1:8000'
    try:
        print('Triggering training...')
        status, body = post(f'{base}/v1/generate/train', {'n_candidates': 6})
        print('train status', status)
        print(body)
    except Exception as e:
        print('Training failed:', e)
        sys.exit(1)

    try:
        print('\nCalling predict...')
        status, body = post(f'{base}/v1/predict/', {'smiles_list': ['CCO', 'c1ccccc1']})
        print('predict status', status)
        print(body)
    except Exception as e:
        print('Predict failed:', e)
        sys.exit(2)
